observability/prometheus-backend/src/main/scala/sttp/client3/prometheus/PrometheusBackend.scala

```diff
@@ -90,7 +143,7 @@ object PrometheusBackend {
       cache.getOrElseUpdate(collectorRegistry, new ConcurrentHashMap[String, T]())
     }
 
-  type RequestCollectors = (Option[Histogram.Timer], Option[Gauge.Child])
+  final class RequestCollectors(val maybeTimer: Option[Histogram.Timer], val maybeGauge: Option[Gauge.Child])
```

Comment:
maybe a `case class`? also, the optionality is already expressed in the types, so I'd just use `timer` / `gauge`

Reply:
Side comment: I tend to avoid `case` if there's no need for equality, i.e. `class` w/ `val`'s meets the bare minimum. 

But, adding `case` is a trivial addition, so I'm happy to add.

Reply:
Done - https://github.com/softwaremill/sttp/pull/1651/commits/7870222104e55bde081e9d703608568a124ff04d